src/main/java/com/gotocompany/depot/maxcompute/MaxComputeSinkFactory.java

```diff
@@ -67,10 +67,12 @@ public Sink create() {
                 messageParser,
                 partitioningStrategy,
                 maxComputeSinkConfig,
-                sinkConfig
+                sinkConfig,
+                new Instrumentation(statsDReporter, RecordDecorator.class),
+                maxComputeMetrics
         );
         ProtoMessageRecordConverter protoMessageRecordConverter = new ProtoMessageRecordConverter(recordDecorator, maxComputeSchemaCache);
-        return new MaxComputeSink(maxComputeClient, protoMessageRecordConverter,
+        return new MaxComputeSink(maxComputeClient.createInsertManager(), protoMessageRecordConverter,
                 new Instrumentation(statsDReporter, MaxComputeSink.class), maxComputeMetrics);
```

Comment:
pass the StatsDReporter and let the concerned class instantiate the instrumentation